File: backend/models/stockLog.go
package models

import "time"

type StockLog struct {
	ID         uint      `gorm:"primaryKey" json:"ID"`
	StockID    uint      `gorm:"not null" json:"stock_id"`
	Type       string    `gorm:"not null" json:"type"`
	Quantity   float64   `gorm:"not null" json:"quantity"`
	Note       string    `json:"note"`
	SupplierID *uint     `json:"supplier_id"`
	Stock      Stock     `json:"-"`
	Supplier   *Supplier `json:"Supplier,omitempty"`
	CreatedAt  time.Time `json:"CreatedAt"`
}

File: frontend/features/inventory/models/stockLog.ts
export interface StockLog {
  ID: number;
  stock_id: number;
  type: 'IN' | 'OUT' | 'ADJUST';
  quantity: number;
  note: string;
  supplier_id?: number;
  Supplier?: { ID: number; name: string };
  Stock?: {
    Item: { name: string; unit: string };
    Warehouse: { name: string };
  };
  CreatedAt: string;
}
